lms/validation/authentication/_launch_params.py

```diff
@@ -46,20 +46,35 @@ class LaunchParamsAuthSchema(PyramidRequestSchema):
     def __init__(self, request):
         super().__init__(request)
         self._launch_verifier = request.find_service(name="launch_verifier")
+        self._application_instance_service = request.find_service(
+            name="application_instance"
+        )
 
     def lti_user(self):
         """
-        Return an models.LTIUser from the request's launch params.
+        Return a models.LTIUser from the request's launch params.
 
         :raise ValidationError: if the request isn't a valid LTI launch request
 
         :rtype: LTIUser
         """
         kwargs = self.parse(location="form")
 
+        try:
```

Comment:
`LaunchParamsAuthSchema` is really the only place where LTIUser are created from scratch.

The rest of classes in `authentication/` `BearerTokenSchema` and `OAuthCallbackSchema` serialize and un-serialize the one created in `LaunchParamsAuthSchema`.

Reply:
Yep, this is a weird location to be doing all this... but it is how it is. The other way to go would be to do this afterwards in the security policy.